extensions/reflect/src/record.ts

```diff
@@ -0,0 +1,32 @@
+import { Toast, closeMainWindow, open } from "@raycast/api";
+import { runAppleScript } from "run-applescript";
+import { checkReflect, reflectDownload } from "./helpers/reflect";
+
+export default async () => {
+  const reflectInstalled = await checkReflect();
+  if (!reflectInstalled) {
+    const toast = new Toast({
+      title: "Reflect not found",
+      style: Toast.Style.Failure,
+    });
+
+    toast.show();
+    toast.message = "Press ⌘ + D to download";
+    toast.primaryAction = {
+      title: "Download",
+      shortcut: {
+        modifiers: ["cmd"],
+        key: "d",
+      },
+      onAction: async () => await open(reflectDownload),
+    };
+    return false;
+  }
+
+  await closeMainWindow();
+  await runAppleScript(`
+    tell application "Reflect" to activate
+    tell application "System Events" to tell process "Reflect" to ¬
+    keystroke "r" using {command down, shift down}
+  `);
```

Comment:
It's better to add the try-catch to handle errors gracefully.

Reply:
Do you have an example at hand?

Reply:
You can simply wrap them all.

```js
try {
	// ...
} catch (error) {
	if (error instanceof Error) {
		// ... showToast
	}
}
```

Reply:
Checkout new commit